File: Assessment/Bytes/src/app/Service/product.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Product } from '../model/product.model';
import { environment } from '../Environment/environment';
import { Observable } from 'rxjs/internal/Observable';

@Injectable({
  providedIn: 'root'
})
export class ProductService {

  private productUrl = environment.productUrl;
  constructor(private httpClient: HttpClient) {}

  

  postUser(product: FormData): Observable<any> {
    return this.httpClient.post(`${this.productUrl}/addNewProduct`,product);
  }


  getAllProducts():Observable<any>{
    return this.httpClient.get(`${this.productUrl}/getAllProducts`)
  }

  deleteProduct(productId:number):Observable<any>{
    return this.httpClient.delete(`${this.productUrl}/deleteProduct/${productId}`)
  }

  getProductById(productId:number){
    return this.httpClient.get(`${this.productUrl}/getProductById/${productId}`)
  }

  updateProduct(product: FormData): Observable<any> {
    return this.httpClient.post(`${this.productUrl}/updateproduct`,product);
  }

}

File: Assessment/errs/src/main/java/com/bytes/errs/service/ProductService.java
package com.bytes.errs.service;

import java.util.List;
import java.util.Set;

import org.springframework.stereotype.Service;

import com.bytes.errs.model.Image;
import com.bytes.errs.model.Product;

@Service
public interface ProductService {

	Product addNewProduct(Product product);

	List<Product> getAllProducts();

	void deleteProduct(Long productId);


	Product getProductById(Long productId);

	Product updateProduct(Long productId, Product updatedProduct, Set<Image> updatedImages);

}
